crates/prover-engine/src/lib.rs

```diff
@@ -90,6 +93,9 @@ impl ProverEngine {
         self
     }
 
+    /// Starts the prover engine.
+    /// 
+    /// NOTE: This function can only be called synchronously and will block the calling thread.
```

Comment:
This could be detected at runtime, at least to an extent, with something like:
```rust
anyhow::ensure!(
    tokio::runtime::Handle::current().is_err(),
    StartingProverInRuntimeError,
);
```

This will detect whether Tokio runtime is running. It will not detect if *a runtime* is running in general.